x-pack/plugins/endpoint/public/applications/endpoint/store/alerts/middleware.ts

```diff
@@ -8,14 +8,15 @@ import { AlertResultList, AlertData } from '../../../../../common/types';
 import { AppAction } from '../action';
 import { MiddlewareFactory, AlertListState } from '../../types';
 import { isOnAlertPage, apiQueryParams, hasSelectedAlert, uiQueryParams } from './selectors';
+import { cloneHttpFetchQuery } from '../../../../common/clone_http_fetch_query';
 
 export const alertMiddlewareFactory: MiddlewareFactory<AlertListState> = coreStart => {
   return api => next => async (action: AppAction) => {
     next(action);
     const state = api.getState();
     if (action.type === 'userChangedUrl' && isOnAlertPage(state)) {
       const response: AlertResultList = await coreStart.http.get(`/api/endpoint/alerts`, {
-        query: apiQueryParams(state),
+        query: cloneHttpFetchQuery(apiQueryParams(state)),
```

Comment:
The `http.get` function won't accept our `Immutable` type. I'm not sure if this is the best approach long term.
Basically, we require that the return value of `apiQueryParams` not be modified, and `.get` isn't promising that it won't modify them. Therefore we clone it.

Reply:
Seems okay to me.